app/mailers/dossier_mailer.rb

```diff
@@ -183,7 +185,7 @@ def notify_transfer(transfer)
   protected
 
   def prevent_perform_deliveries
-    if params[:commentaire]&.discarded? || params[:dossier]&.skip_user_notification_email?
+    if params[:commentaire]&.discarded? || (params[:commentaire]&.dossier || params[:dossier])&.skip_user_notification_email?
```

Comment:
j'ai mis trop de temps à comprendre, donc je propose : 

```suggestion
    commentaire = params[:commentaire]
    dossier = commentaire&.dossier || params[:dossier]

    if commentaire&.discarded? || dossier&.skip_user_notification_email?
```